src/main/java/com/example/springresttask/service/FileService.java

```diff
@@ -0,0 +1,104 @@
+package com.example.springresttask.service;
+
+import com.example.springresttask.model.UploadedFile;
+import com.example.springresttask.repository.FileRepository;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@Service
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+@RequiredArgsConstructor
+@Slf4j
+public class FileService {
+    FileRepository fileRepository;
+    Path root = Path.of(System.getProperty("user.dir") + "\\src\\main\\resources\\uploadFiles");
```

Comment:
Move this to a separate class, make static and final then use it in FileService and all other possible places